[Ee Suan] iP

ee-suan opened the pull request:
# Duke
> “Edit your life frequently and ruthlessly. It’s your masterpiece after all.” – Nathan W. Morris ([source](https://dansilvestre.com/productivity-quotes))

Duke frees your mind of having to remember things you need to do. It's,

- text-based
- easy to learn
- ~~FAST~~ *SUPER* FAST to use

All you need to do is,

1. download it from [here](https://github.com/ee-suan/ip).
2. double-click it.
3. add your tasks.
4. let it manage your tasks for you 🤩 

And it is **FREE!**

Features:

- [x] Managing tasks
- [x] Managing deadlines 
- [ ] Reminders (coming soon)


#### If you're a Java programmer, you can use it to practice Java too. Here's the `main` method:

```java
public class Main {
    public static void main(String[] args) {
        Application.launch(MainApp.class, args);
    }
}
```

Commit:
Add Gradle support

ee-suan committed:
Add Level-1

ee-suan committed:
Add Level-2

ee-suan committed:
Abstract greet() and exit()

ee-suan committed:
Add Task class

ee-suan committed:
Add Level-3

ee-suan committed:
Add support for todos, deadlines, and events (Level-4)

ee-suan committed:
Add I/O redirection testing

ee-suan committed:
Handle some errors

ee-suan committed:
Abstracted Duke's reply message

ee-suan committed:
Add Level-6

ee-suan committed:
Add more error handling with exceptions

ee-suan committed:
Add support for saving file

ee-suan committed:
Add support for dates

ee-suan committed:
Merge branch 'branch-Level-8'

ee-suan committed:
Add more OOP

ee-suan committed:
Organise classes into package

ee-suan committed:
Merge branch 'add-gradle-support' of https://github.com/nus-cs2103-AY2223S1/ip

ee-suan committed:
Add JUnit tests

ee-suan committed:
adjust file path

ee-suan committed:
Add JavaDoc comments

ee-suan committed:
Update code to comply with coding standard

ee-suan committed:
Add support for find

ee-suan committed:
Merge branch 'A-CodingStandard'

# Conflicts:
#	src/main/java/duke/Ui.java

Resolution: keep StringBuilder

ee-suan committed:
Merge branch 'Level-9'

# Conflicts:
#	src/main/java/duke/Duke.java
#	src/main/java/duke/TaskList.java
#	src/main/java/duke/Ui.java

Resolved whitespaces manually

sjoann commented on src/main/java/duke/Ui.java:
```diff
@@ -18,12 +17,12 @@ public void exit() {
 
     public void printList(TaskList taskList) {
-        String message = "Here are the tasks in your list:";
+        StringBuilder message = new StringBuilder("Here are the tasks in your list:");
```
Good job in using Stringbuilder :)

sjoann commented on src/main/java/duke/Parser.java:
```diff
@@ -49,6 +49,11 @@ public void parse() {
                         duke.handleDelete(index);
                         break;
                     }
+                    case "find": {
```
I like that you use switch case for this

sjoann commented on src/test/java/duke/DeadlineTest.java:
```diff
@@ -0,0 +1,33 @@
+public class DeadlineTest {
+
+    /**
+     * Test Deadline with valid input
```
maybe the first sentence could start with Tests instead of Test (eg. Returns instead of Return)

sjoann commented on src/main/java/Parser.java:
```diff
@@ -0,0 +1,60 @@
+        while (!input.equals("bye")) {
+            try {
+                switch (input) {
+                    case "list":
```
I think there is no indentation for case clauses

sjoann reviewed (commented):
Overall great work ! The code is easy to read and has JavaDocs

xinbei-shen commented on src/main/java/duke/Duke.java:
```diff
@@ -0,0 +1,161 @@
+    }
+
+    private Task createDeadline(String input) throws DukeException {
+        int end = input.indexOf('/');
+        String description = input.substring(9, end - 1);
+        String by = input.substring(end + 4);
```
perhaps this can be done by the parser? since it is the parser that makes sense of the input
similar for createEvent

prit3010 commented on src/main/java/duke/Parser.java:
```diff
@@ -0,0 +1,79 @@
+                        input = sc.nextLine().trim();
+                        duke.find(input);
+                        break;
+                    }
```
Good use of switch case but Switch and Case should not be indented.
Eg.
```
switch (condition) {
case ABC:
    statements;
    // Fallthrough
case DEF:
    statements;
    break;
case XYZ:
    statements;
    break;
default:
    statements;
    break;
}
```

prit3010 commented on src/main/java/duke/TaskList.java:
```diff
@@ -0,0 +1,123 @@
+ * TaskList class containing list of tasks and their operations.
+ */
+public class TaskList {
+    private ArrayList<Task> list;
```
Perhaps this  `list` variable can be more descriptive like `tasks`?

xinbei-shen commented on src/main/java/duke/Deadline.java:
```diff
@@ -0,0 +1,45 @@
+import java.time.format.DateTimeParseException;
+
+/**
+ * Deadline class which is a subclass of Event.
```
perhaps this can be more descriptive? for example "This class encapsulates a deadline that contains..."
same for Event and ToDo

xinbei-shen commented on src/main/java/duke/Task.java:
```diff
@@ -0,0 +1,59 @@
+/**
+ * Abstract Task class.
+ */
+public abstract class Task {
```
i like how you used an appropriate class for Task!

xinbei-shen commented on src/main/java/duke/TaskList.java:
```diff
@@ -0,0 +1,123 @@
+    public Task delete(int i) throws DukeException {
+        int size = this.size();
+        if (size == 0) {
+            throw new DukeException("☹ OOPS!!! You do not have any task right now.");
```
Perhaps you could use unicode for the sad face for example `char sadFace = '\u2639'` because it shows up as scrambled code for me in my terminal

xinbei-shen reviewed (commented):
Very clean code with good documentation for most classes and methods!! good job 😃

prit3010 commented on src/main/java/duke/Deadline.java:
```diff
@@ -0,0 +1,45 @@
+        return String.format("D | %s | %s | %s", status, this.description, this.by);
+    }
+
+    /**
+     * @inheritDoc
+     */
```
Good Use of `@inheritDoc`

prit3010 commented on src/main/java/duke/Storage.java:
```diff
@@ -0,0 +1,94 @@
+        this.file = new File(filePath);
+    }
+
+    /**
+     * Saves TaskList to a file.
+     *
+     * @param tasks A TaskList object containing lists of tasks.
+     */
```
JavaDocs is well done!

prit3010 commented on src/main/java/duke/Ui.java:
```diff
@@ -0,0 +1,80 @@
+
+    public void dukeReply(String message) {
+        String lineBreak = "-------------------------------------------------";
+        String reply = String.format("%s\n%s\n%s", lineBreak, message, lineBreak);
```
Variable should not be a verb. Perhaps you can use `replyMessage`

prit3010 reviewed (commented):
Code is very good! Some minor errors here but overall, follows the coding standards very closely. Good Job.

ee-suan committed:
Add GUI

ee-suan committed:
Add Checkstyle

ee-suan committed:
Fix dialogue box size and clean up code

ee-suan committed:
Add assertions

ee-suan committed:
Fix coding style

ee-suan committed:
Enable Assertions

ee-suan committed:
Merge pull request #1 from ee-suan/branch-A-Assertions

Branch a assertions

ee-suan committed:
Merge branch 'master' into branch-A-CodeQuality

# Conflicts:
#	src/main/java/duke/Storage.java

ee-suan committed:
Merge pull request #2 from ee-suan/branch-A-CodeQuality

Branch a code quality

ee-suan committed:
Add support for BetterSearch

ee-suan committed:
Merge branch 'branch-C-BetterSearch'

ee-suan committed:
Add support for input error handling

The syntax allowed for commands are strict, especially deadline.
If not handled well the program would crash at bad inputs.

Let's split the input string and check each section against what is
allowed. An exception is throw, then caught to display error message
to user if command is invalid.

ee-suan committed:
Change the name used in introduction message

ee-suan committed:
Update welcome message

ee-suan committed:
Add Ui image to docs

ee-suan committed:
Add a User Guide

ee-suan committed:
Update README.md
